src/keyword_vpd_parser.cpp

```diff
@@ -48,12 +53,19 @@ types::VPDMapVariant KeywordVpdParser::parse()
             throw(DataException("Invalid Small resource type End"));
         }
 
+        types::BinaryVector::const_iterator l_checkSumEnd = m_vpdIterator;
+        validateChecksum(l_checkSumStart, l_checkSumEnd);
+
+        checkNextBytesValidity(constants::TWO_BYTES);
+        std::advance(m_vpdIterator, constants::TWO_BYTES);
+
         // Check vpd end Tag.
         if (*m_vpdIterator != constants::KW_VPD_END_TAG)
         {
             throw(DataException("Invalid Small resource type."));
         }
-        return kwValMap;
+
+        return l_kwValMap;
     }
     catch (const DataException& ex)
```

Comment:
Though unrelated to this commit but as we are refactoring, I would like you to take this change in this PR.
This catch here does not make any sense as it is just logging and re-throwing the same exception.
Either:
a) Handle exception here itself or,
b) Handle it at the calling place only.

Reply:
Handling it at the calling place.